manifests/repos.pp

```diff
@@ -22,19 +37,62 @@
       $package_key   = $docker::package_key
       $package_repos = $docker::package_repos
 
+      if ( $facts['os']['distro']['id'] == 'Debian' and versioncmp($facts['os']['distro']['release']['major'],'10') >= 0 ) or ( $facts['os']['distro']['id'] == 'Ubuntu' and versioncmp($facts['os']['distro']['release']['major'],'22') >= 0 ) { # lint:ignore:140chars
+        # fix deprecated apt-key warnings
+        if $curl_ensure {
+          ensure_packages(['curl'])
+        }
+        if $gpg_ensure {
+          ensure_packages(['gpg'])
+        }
+        if $keyring_force_update {
+          exec { 'Remove Docker-GPG-Key':
```

Comment:
This is not idempotent and will be executed any time when puppet agent is running.

Reply:
You are absolutly right.  
I will take a look if there is a better solution with apt::key.

Reply:
Since I couldn't figure out a 'nicer solution' I removed the 'keyring_force_update' functionality with 75b70d8bb8671530710cfe48c152dc2f6abc00e8.